src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs

```diff
@@ -295,6 +298,25 @@ public class UpgradeOrganizationPlanCommand : IUpgradeOrganizationPlanCommand
         return new Tuple<bool, string>(success, paymentIntentClientSecret);
     }
 
+    private static string GetUpgradePath(ProductType oldProductType, ProductType newProductType)
+    {
+        var upgradePaths = new Dictionary<(ProductType, ProductType), string>
+        {
+            [(ProductType.Free, ProductType.Families)] = "2-person org → Families",
+            [(ProductType.Free, ProductType.TeamsStarter)] = "2-person org → Teams Starter",
+            [(ProductType.Free, ProductType.Teams)] = "2-person org → Teams",
+            [(ProductType.Free, ProductType.Enterprise)] = "2-person org → Enterprise",
+            [(ProductType.Families, ProductType.TeamsStarter)] = "Families → Teams Starter",
+            [(ProductType.Families, ProductType.Teams)] = "Families → Teams",
+            [(ProductType.Families, ProductType.Enterprise)] = "Families → Enterprise",
+            [(ProductType.TeamsStarter, ProductType.Teams)] = "Teams Starter → Teams",
+            [(ProductType.TeamsStarter, ProductType.Enterprise)] = "Teams Starter → Enterprise",
+            [(ProductType.Teams, ProductType.Enterprise)] = "Teams → Enterprise"
+        };
```

Comment:
⚠️ Could this mapping be pulled into a static variable? That'd prevent it from being initialized every time this method is invoked.

🤔 This might also be better structured as a lookup for each member of the tuple.

```csharp
var memberships = new Dictionary<ProductType> {
  [ProductType.Free] = "2-person org",
  [ProductType.Families] = "Families",
  [ProductType.TeamsStarter] = "Teams Starter",
  // ...
};

if(memberships.TryGetValue(oldProductType, out oldDescription) &&
   memberships.TryGetValue(newProductType, out newDescription))
{
  return $"{oldDescription} → {newDescription}";
} else {
  return null;
}
```

Reply:
You are right, I have the refactored the code  as suggested